### ISSUE TITLE
Split an ONNX Model

### ISSUE THREAD

**Issue by thestarivore** [onnx/onnx#3770]
Hi,

### Question
Is it possible to use the split operation to split an ONNX Model into two models and run them separately?  If so what is the correct way to do it?
Thank you in advance.

### Further information
I must basically split the LeNET DNN model into two models and run them on different machines. So after getting the results of the first half on the model I must use it as input for the second half and run it to get the same result as if I were to run the whole model. But I can't get split them correctly.

The model I use is the following:
`model = keras.Sequential()`
`model.add(layers.Conv2D(filters=6, kernel_size=(3, 3), activation='relu', input_shape=(32,32,1)))`
`model.add(layers.AveragePooling2D())`
`model.add(layers.Conv2D(filters=16, kernel_size=(3, 3), activation='relu'))`
`model.add(layers.AveragePooling2D())`
`model.add(layers.Flatten())`
`model.add(layers.Dense(units=120, activation='relu'))`
`model.add(layers.Dense(units=84, activation='relu'))`
`model.add(layers.Dense(units=10, activation = 'softmax'))`

And I have trained it and exported it as ONNX. 
Then I start from this model trying to create the two halves:

- for the first one I insert an output at a middle point (and run the model from the original input to the newly created output);
- for the second model I insert an input at a middle point (and run the model from this newly created input to the original output);

My first attempt is the following (I also attached the exported images of the graphs):

First model:
`node_identity = onnx.helper.make_node('Identity', inputs=['sequential/dense_1/Relu'], outputs=['output2'], name='identity_node')`
`onnx_model.graph.node.append(node_identity)`
`onnx_model.graph.output.append(onnx_model.graph.output[0])`
`onnx_model.graph.output[1].name = "output2"`

Second model:
`onnx_model.graph.input.append(onnx_model.graph.input[0])`
`onnx_model.graph.input[1].name = "middle_input"`

`for i in range(len(onnx_model.graph.node)):`
`	print(onnx_model.graph.node[i].name)`
`	if onnx_model.graph.node[i].name == "sequential/dense_2/MatMul":`
`    		onnx_model.graph.node[i].input[0] = "middle_input"`

**Model name** (*mnist, LeNet-5 network architecture*):  
![graph_modified2 dot](https://user-images.githubusercontent.com/22894195/136956409-51342de8-8b23-4e7b-9439-74803c7e84b2.png)
![graph_modified dot](https://user-images.githubusercontent.com/22894195/136959771-affd016c-4d21-4134-9283-a5ab80645d3e.png)

**Comment by thestarivore** [comment]
> If you know what variables to use as the outputs/inputs of each model, you can likely use this utility to do it: https://github.com/onnx/onnx/blob/master/docs/PythonAPIOverview.md#extracting-sub-model-with-inputs-outputs-tensor-names ... you will have to invoke it twice, once to get the first model and once to get the second model.

Thank you very much for your reply, the **extract_model** utility seems indeed to be what I have been looking for. But I'm not sure how to get the input and output tensors of the model I have produced. If I list the whole onnx file I can see the nodes, inputs, outputs, initializers but no tensor..

[onnx_model.txt](https://github.com/onnx/onnx/files/7353416/onnx_model.txt)

Could you please give me some indications on how to get the correct tensor names? Thank you in advance.

**Comment by vetsasai** [comment]
@gramalingam @thestarivore 
Do you know a smart way to identify the correct point in the graph to split the model (to minimize the dependancy between two submodels).? I am trying to split Gemma3 1B model, but there seem to be lot of dependancies between subgraphs, if I pick the point manually. Your advice would be really helpful (if there are some tools/tutorials/articles that can help me do this.)
